src/components/Regridder.vue

```diff
@@ -0,0 +1,232 @@
+        },
+        modes:  Object.assign(MapboxDraw.modes, {
+            draw_rectangle: DrawRectangle,
+        } as any),
+        defaultMode: 'draw_rectangle',
+        })
+
+        this.mapObject.addControl(this.draw)
+
+        this.draw.changeMode("draw_rectangle")
+
+        this.mapObject.on('draw.create', this.select)
+
+        const trashElement = document.querySelector(".mapbox-gl-draw_trash")
+        if (trashElement !== null) {
+            trashElement.addEventListener("click", () => {
+                this.draw.deleteAll()
+                this.refreshDownloadControl(undefined)
+            })
+        }
+        const drawCombine = document.querySelector(".mapbox-gl-draw_polygon")
+        if (drawCombine !== null) {
+            drawCombine.addEventListener("click", () => {
+                this.draw.deleteAll()
+                this.draw.changeMode("draw_rectangle")
+            })
+        }
+        // Create the DownloadControl initially
+        this.downloadControl = new DownloadControl(undefined, this)
+        this.mapObject.addControl(this.downloadControl)
+    }
+
+    mounted() {
+    const map = this.getMap()
+    if (map && map.loaded()) {
+        this.mapObject = map
+        this.isInitialized = true
+        this.addToMap()
+    }
+    }
+    downloadClicked() {
+    const dx = parseFloat(this.dx)
+    const dy = parseFloat(this.dy)
+
+    if (!isNaN(dx) && !isNaN(dy) && dx > 0 && dy > 0 && this.bbox !== undefined)  {
+        const workflowId = 'Transformation_ASA_Grid_Wind'
+        const xMin = this.bbox[0]
+        const yMin = this.bbox[1]
+        const xMax = this.bbox[2]
+        const yMax = this.bbox[3]
+        const xCellSize = dx
+        const yCellSize = dy
+        const startTime = this.firstValueTime
+        const endTime = this.lastValueTime
+        // todo: change this with fews-pi-requests
+        const apiUrl = `${this.baseUrl}rest/fewspiservice/v1/processdata?workflowId=${workflowId}&xMin=${xMin}&yMin=${yMin}&xMax=${xMax}&yMax=${yMax}&xCellSize=${xCellSize}&yCellSize=${yCellSize}&startTime=${startTime}&endTime=${endTime}`
+        this._downloadNetCDF(apiUrl)
+        this.downloadDialog = false
+    } else {
+        // Handle invalid input
+        this.closeDownloadDialog()
+        this.openErrorDialog('Invalid input. Please enter valid values for dx and dy.')
+    }
+    }
+
+    closeDownloadDialog() {
+    this.downloadDialog = false
+    }
+
+    openErrorDialog(errorMessage: string) {
+    this.errorMessage = errorMessage
+    this.errorDialog = true
+    }
+
+    closeErrorDialog() {
+    this.errorDialog = false
+    // reset error message
+    this.errorMessage = 'Select an area on the map before downloading the data.'
+    }
+
+    _downloadNetCDF(apiUrl: string) {
+        let fileName = 'openarchive-netcdf.nc'
+        // Use the Fetch API to make the request, change later to fews-pi-requests
+        fetch(apiUrl)
+        .then((response) => {
```

Comment:
Use `async` function and `await`? It's a bit more easy to read and more consistent with the rest of `fews-web-oc`.